src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -136,7 +157,7 @@ private Task<bool> NominateProjectAsync(string projectUniqueName, IVsProjectRest
             catch (Exception e)
             {
                 _logger.LogError(e.ToString());
-                return Task.FromResult(false);
+                throw;
```

Comment:
Why are we throwing?

Reply:
`_logger` just goes to the VS activity log, but we have no way of knowing how often customers are actually getting this as it's otherwise unreported. Once this PR is merged, the only known way of reaching this exception handler will no longer be possible, so I thought it'd be reasonable to allow it to generate Watson reports again.

Do you have another suggestion? Should I make it generate a telemetry event?

Reply:
I don't think we should have Watson reports for users errors maybe?

Reply:
As far as the activity log goes, it's only for us...customers shouldn't see it, the customers will see the issues in the error window.

As far as a telemetry event goes, I'll leave that at your discretion. Don't know if we have a guidance for these things, beyond fire and forget events.

Reply:
Indeed we shouldn't have Watson reports for user errors, but the new nomination error handling should catch user errors gracefully. If an exception somehow gets to this comment's catch block, that means that graceful nomination error handling didn't work, and the user error will not be reported in the assets file.

In other words, there was an application error in being unable to handle the root cause gracefully.

Reply:
I'd rather us handle it than rely on some other teams' infrastructure to report this.

Reply:
@zivkan before I review again, can you look at the above, specifically 

> I'd rather us handle it than rely on some other teams' infrastructure to report this.

If we think it's important enough, we should track is ourselves rather than rely on partner reporting.

Reply:
@nkolev92 updated to report errors without creating watson crashes.